src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -64,16 +58,53 @@ public BlobFeedAction(string accountName, string accountKey, string containerNam
 
             using (var clientThrottle = new SemaphoreSlim(this.MaxClients, this.MaxClients))
             {
-                await Task.WhenAll(items.Select(item => PushItemToFeed(item, relativePath, clientThrottle)));
+                await Task.WhenAll(items.Select(item => PushItemToFeed(item, relativePath, clientThrottle, allowOverwrite)));
             }
             return !Log.HasLoggedErrors;
         }
 
-        public async Task<bool> PushItemToFeed(string item, string relativePath, SemaphoreSlim clientThrottle)
+        public async Task<bool> PushItemToFeed(string item, string relativePath, SemaphoreSlim clientThrottle, bool allowOverwrite)
         {
-            string uploadPath = feed.CalculateBlobPath(item, relativePath);
-            List<string> listAzureBlobs = await ListAzureBlobs.ListBlobs(Log, feed.AccountName, feed.AccountKey, feed.ContainerName, feed.CalculateRelativeUploadPath(item, relativePath));
-            await UploadAsync(CancellationToken, item, uploadPath, clientThrottle, listAzureBlobs.Contains(uploadPath));
+            try
+            {
+                Tuple<string, NuGetVersion> nugetPackage = feed.GetNugetPackageVersion(item);
+                string uploadPath = feed.CalculateBlobPath(item, relativePath);
+                string searchPath = feed.CalculateRelativeUploadPath(item, relativePath);
+
+                List<string> listAzureBlobs = await ListAzureBlobs.ListBlobs(Log, feed.AccountName, feed.AccountKey, feed.ContainerName, uploadPath);
+                if (listAzureBlobs.Contains(uploadPath) && !allowOverwrite)
```

Comment:
Only do the list if allowOverwrite is false. Should just do a file exists instead of list.